apps/readingchallenge/reading_challenge.star

```diff
@@ -0,0 +1,158 @@
+        ])
+    elif progress == 7:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = FIVE_BOOK_ICON, height = 13, width = 20),
+            render.Padding(child = render.Image(src = ONE_CLOSED_BOOK_ICON, height = 15, width = 10), pad = (3, 0, 0, 0)),
+            render.Padding(child = render.Image(src = ONE_OPEN_BOOK_ICON, height = 15, width = 25), pad = (2, 0, 0, 0)),
+        ])
+
+    elif progress == 8:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = THREE_BOOK_ICON, height = 13, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+        ])
+    elif progress == 9:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = THREE_BOOK_ICON, height = 13, width = 20),
+            render.Image(src = THREE_BOOK_ICON, height = 13, width = 20),
+            render.Image(src = THREE_BOOK_ICON, height = 13, width = 20),
+        ])
+    elif progress == 10:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+        ])
+    elif progress == 11:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+            render.Padding(child = render.Image(src = ONE_CLOSED_BOOK_ICON, height = 15, width = 12), pad = (1, 0, 0, 0)),
+        ])
+
+    elif progress == 12:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = FIVE_BOOK_ICON, height = 12, width = 18),
+            render.Image(src = THREE_BOOK_ICON, height = 22, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 16, width = 10),
+        ])
+    else:
+        images = render.Row(main_align = "center", cross_align = "end", expanded = True, children = [
+            render.Image(src = FIVE_BOOK_ICON, height = 15, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 14, width = 20),
+            render.Image(src = FIVE_BOOK_ICON, height = 12, width = 18),
+            render.Image(src = FIVE_BOOK_ICON, height = 16, width = 10),
+        ])
+
+    return images
+
+def main(config):
+    CHALLENGE_ID = secret.decrypt("AV6+xWcE/k5WZk3zYbxM/fuI6z4DJ5oQUNrHWbIu8HBOyeubTAzYcB9DxEMAWRoxDcryQFohadBlZneiSq94urEiiG53bpXexcibIvOyKOSH3Vi3qEYaY1vVXUxHQOmyk5odoCU9UtLX850GqOI=")
+
+    if CHALLENGE_ID == None:
+        CHALLENGE_ID = config.str("user_challenge_id", "0")
+
+    progress_cached = cache.get("progress")
```

Comment:
The cache is shared by all users of the app. This will retrieve and use the most recently written progress record, regardless of what user create it. You probably want to include the challenge_id in the cache key.

Reply:
That makes sense, thanks for the clarification